test-crates/duchess-java-tests/build.rs

```diff
@@ -0,0 +1,80 @@
+use std::{
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use walkdir::WalkDir;
+
+// These two control where this script looks for source and corresponding class files
+const SOURCE_PATH: &str = "java";
+const TARGET_PATH: &str = "target";
+
+fn main() -> std::io::Result<()> {
+    // Rerun java build if any source file changes, but then we'll check each file individually below
+    println!("cargo:rerun-if-changed={}", SOURCE_PATH);
+
+    let target_dir = Path::new(TARGET_PATH);
```

Comment:
This `test-crates/duchess-java-tests/target` folder will not be deleted by any `cargo clean`. What about placing the target dir in `../target/duchess-java-tests/`, so that it'll go away with a `cd test-crates && cargo clean`?